packages/p/packagekit/files/0001-data-Tweak-systemd-network-targets-to-get-offline-up.patch

```diff
@@ -0,0 +1,45 @@
+From f8364104fa7266735fb08e50d988f6e80acadf7a Mon Sep 17 00:00:00 2001
+From: Joey Riches <josephriches@gmail.com>
+Date: Sun, 22 Oct 2023 10:31:23 +0100
+Subject: [PATCH 1/1] data: Tweak systemd network targets to get offline update
+ working reliably
+
+Solus, downstream only.
+
+Even when an offline transation is pre-downloaded pk still wants network :(
+
+We were hitting an issue where pk offline update would try to start
+the transaction before networking was up.
+---
+ data/packagekit-offline-update.service.in | 1 +
+ data/packagekit.service.in                | 2 +-
+ 2 files changed, 2 insertions(+), 1 deletion(-)
+
+diff --git a/data/packagekit-offline-update.service.in b/data/packagekit-offline-update.service.in
+index 23096058f..b2c19cd03 100644
+--- a/data/packagekit-offline-update.service.in
++++ b/data/packagekit-offline-update.service.in
+@@ -7,6 +7,7 @@ After=sysinit.target dbus.socket systemd-journald.socket system-update-pre.targe
+ Before=shutdown.target system-update.target
+ # See packagekit.service
+ ConditionPathExists=!/run/ostree-booted
++Wants=network-online.target
```

Comment:
As I mentioned in Matrix, I think there is something wrong with our implementation if this is required. Fedora doesn't have any similar changes and their implementation works fine there. In this respect they shouldn't be substantially different than what we are doing.

Reply:
> I think there is something wrong with our implementation

Implementation of our packagekit backend or our systemd stuff?

Reply:
Okay i noticed it is still funky even with this patch.

Reply:
Well the question is why is it needed even in the first place? Why does `packagekit-offline-update` depend on packagekit.service in the first place? Is it because it wants the service to actually be started, or is it just checking for the socket to be active? I imagine that it should be starting all necessary services itself, which is why it needing `packagekit.service` is a red flag to me. I think if we can determine _why_ it needs packagekit.service then we can fix that part and this patch will no longer be necessary (and it'll probably fix the funkiness you're seeing).

Reply:
packagekitd is a dbus activated daemon that will wake up as needed and shut itself down after x amount of time

Reply:
Anything using packagekit will use it